llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -142,20 +136,31 @@ bool isEntryPoint(const Function &F, bool EmitOnlyKernelsAsEntryPoints) {
   return false;
 }
 
+bool isESIMDFunction(const Function &F) {
+  return F.getMetadata(ESIMD_MARKER_MD) != nullptr;
+}
+
 // This function makes one or two groups depending on kernel types (SYCL, ESIMD)
-// if SplitEsimd is true. Otherwise, all kernels are collected into one group.
 EntryPointGroupVec
-groupEntryPointsByKernelType(const Module &M, bool SplitEsimd,
-                             bool EmitOnlyKernelsAsEntryPoints) {
+groupEntryPointsByKernelType(const Module &M, bool EmitOnlyKernelsAsEntryPoints,
+                             EntryPointVec *AllowedEntriesVec) {
+  SmallPtrSet<const Function *, 32> AllowedEntries;
+
+  if (AllowedEntriesVec) {
```

Comment:
Why do we need a copy here?

Reply:
This is just an idiom to insert (copy) all elements of the vector into a set for faster lookup below (see line 160)